src/publicodes-state/helpers/safeGetSituation.ts

```diff
@@ -1,18 +1,53 @@
-import { NodeValue, Situation } from '../types'
+import { Situation } from '../types'
+import { dottedNamesMigration } from './dottedNamesMigration'
 
 export const safeGetSituation = ({
   situation,
   everyRules,
 }: {
   situation: Situation
   everyRules: string[]
-}): any =>
-  everyRules
-    .filter((rule: string) => situation[rule] || situation[rule] === 0)
-    .reduce(
-      (accumulator: Record<string, NodeValue>, currentValue: string) => ({
-        ...accumulator,
-        [currentValue]: situation[currentValue],
-      }),
-      {}
-    )
+}): any => {
+  const unsupportedDottedNamesFromSituation = Object.keys(situation).filter(
+    (dottedName) => {
+      // We check if the dotteName is a rule of the model and migrate the rule if it's possible
+      if (!everyRules.includes(dottedName)) {
+        return true
+      }
+      // We check if the value of a dottedName waiting for an answer from "une possiblité"
+      // is defined as a rule `dottedName . value` in the model
+      // and migrate the rule if it's possible
+      if (
+        typeof situation[dottedName] === 'string' &&
+        situation[dottedName] !== 'oui' &&
+        situation[dottedName] !== 'non' &&
+        !everyRules.includes(`${dottedName} . ${situation[dottedName]}`)
+      ) {
+        return true
+      }
+    }
+  )
+
+  const filteredSituation = situation
+
+  unsupportedDottedNamesFromSituation.map((dottedName: string) => {
+    const situationValue = situation[dottedName] as string
+    if (Object.keys(dottedNamesMigration.key).includes(dottedName)) {
+      delete Object.assign(filteredSituation, {
+        [dottedNamesMigration.key[dottedName]]: situationValue,
+      })[dottedName]
+    } else if (
+      Object.keys(dottedNamesMigration.value).includes(dottedName) &&
```

Comment:
```suggestion
      else if(Object.keys(dottedNamesMigration.value)?.[dottedName]?.includes(situationValue)) {
```
Version plus concise ?

Reply:
Je suis pas sur qu'on puisse, il faut vérifier que `situationValue` est dans les clés de `Object.keys(dottedNamesMigration.value?.[dottedName])` or pour accéder aux clés de `dottedNamesMigration.value?.[dottedName]` il faut que ce soit non nul

Reply:
Okay pardon j'ai fait une erreur on peut faire :
```
else if(dottedNamesMigration.value?.[dottedName]?.includes(situationValue))
```